pkg/ccl/backupccl/backup_span_maker_test.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package backupccl
+
+import (
+	"context"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
+	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+	"github.com/cockroachdb/cockroach/pkg/util/rangedesc"
+	"github.com/cockroachdb/cockroach/pkg/util/span"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBackupSpanBoundOnRange(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	tc, sqlDB, _, cleanup := backupRestoreTestSetup(t, singleNode, 1, InitManualReplication)
```

Comment:
Do we need to make a whole server? what would you think about just implementing `rangedesc.Iterator` ? Bonus points for a `rangedesc.TestingIterator(descs... roachpb.RangeDescriptor)` helper for next time?

Reply:
are you asking why we need a whole test _cluster_, as oppose to just a test _server_? Don't we need to the test server to open a sql connection?

Reply:
I believe what dt is getting at is that you can test `makeRequestSpans` without a running test server at all if you passed `makeRequestSpans` something that implements `rangedesc.Iterator` and returned range descriptors that you explicitly specified.